### PR TITLE
Fail at instantiating a packed scene if the root node is instantiated

### PR DESCRIPTION
**Pull request by gyroninja**
Godot corrupted one of my scenes and revealed a bug that crashes godot when it tries to load it. To reproduce it you can use a scene file like:

```
[gd_scene]
[node name="a"]
[node name="b" parent="c"]
```

While this patch fixes the crash, I am not familiar enough with godot to know if this type of file is supposed to be invalid. I also didn't use ERR_FAIL_COND_V_MSG because it don't really understand what the problem is.

### CODE REVIEW

**Comment by RandomShaper** [comment]
That scene file is wrong indeed. The root node must specify a type if it's a base scene, or a base scene if it's a derived scene.

Your patch does takes care of the bad case where there's neither a base scene nor a type. However, I think this would be a better way:

Replace
`// i == 0 is root node. Confirm that it doesn't have a parent defined.`
by this
```
// i == 0 is root node.
ERR_FAIL_COND_V_MSG(n.parent != -1, nullptr, vformat("Invalid scene: root node %s cannot specify a parent node.", snames[n.name]));
ERR_FAIL_COND_V_MSG(n.type == TYPE_INSTANCED && base_scene_idx < 0, nullptr, vformat("Invalid scene: root node %s in an instance, but there's no base scene.", snames[n.name]));
```

I think that doing the sanity check earlier, so the error condition is more explicit, and beside the other one, makes the code clearer.

**Comment by gyroninja** [comment]
Okay I've changed what you requested. I noticed there was a cherrypick:3.3 tag added, should I go ahead and make another PR for that right now?

**Comment by akien-mga** [comment]
> I noticed there was a cherrypick:3.3 tag added, should I go ahead and make another PR for that right now?

No, that's intended for the release manager (me), I'll cherry-pick the changes myself to those branches once merged.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.4.

**Pull request merged**
